samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -113,7 +115,7 @@ private void validateConfig() {
    */
   /* package private */
   JobGraph createJobGraph(ApplicationDescriptorImpl<? extends ApplicationDescriptor> appDesc) {
-    JobGraph jobGraph = new JobGraph(config, appDesc);
+    JobGraph jobGraph = new JobGraph(new MapConfig(config), appDesc);
```

Comment:
why we need to wrap MapConfig on the config?

Reply:
yes that should not exist